### PR TITLE
Add examples

### PR DESCRIPTION
**Pull request by KATTA-00**
## Purpose

- Added two example with these documentations
- Updated a README.md for examples

### CODE REVIEW

File: ballerina/Ballerina.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 distribution = "2201.9.2"
 org = "ballerinax"
 name = "openai.finetunes"
-version = "1.0.5"
+version = "1.0.7"
```
Comment (NipunaRanasinghe) [inline ballerina/Ballerina.toml]: lets revert this version update

File: examples/Sarcastic-bot/Ballerina.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[package]
+org = "kanishka"
+name = "Sports_headline_analyzer"
```
Comment (NipunaRanasinghe) [inline examples/Sarcastic-bot/Ballerina.toml]:
```suggestion
name = "sarcastic_bot"
```

File: examples/Sarcastic-bot/Ballerina.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[package]
+org = "kanishka"
```
Comment (NipunaRanasinghe) [inline examples/Sarcastic-bot/Ballerina.toml]: We usually use the `wso2` as the orgname for the examples
```suggestion
org = "wso2"
```

File: examples/Sarcastic-bot/Ballerina.toml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+distribution = "2201.9.2"
+
+[build-options]
+observabilityIncluded = true
+
+[[dependency]]
+org = "ballerinax"
+name = "openai.finetunes"
+version = "1.0.7"
+repository = "local"
```
Comment (NipunaRanasinghe) [inline examples/Sarcastic-bot/Ballerina.toml]: lets remove this

File: examples/Sarcastic-bot/Dependencies.toml
Comment (NipunaRanasinghe) [inline examples/Sarcastic-bot/Dependencies.toml:1]: Usually we don't commit the `Dependencies.toml` file for the examples. So we can remove this too

File: examples/Sarcastic-bot/main.bal [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+import ballerina/io;
+import ballerina/lang.runtime;
+import ballerinax/openai.finetunes;
+
+configurable string token = ?;
+string serviceUrl = "https://api.openai.com/v1";
+string trainingFileName = "training.jsonl";
+string validationFileName = "validation.jsonl";
+string trainingFilePath = "./data/" + trainingFileName;
+string validationFilePath = "./data/" + validationFileName;
```
Comment (NipunaRanasinghe) [inline examples/Sarcastic-bot/main.bal]:
```suggestion
const SERVICE_URL = "https://api.openai.com/v1";
const TRAINING_FILENAME = "training.jsonl";
const VALIDATION_FILENAME = "validation.jsonl";
const TRAINING_FILEPATH = "./data/" + trainingFileName;
const VALIDATION_FILEPATH = "./data/" + validationFileName;
```

File: examples/Sports-headline-analyzer/Ballerina.toml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+[package]
+org = "kanishka"
+name = "Sports_headline_analyzer"
```
Comment (NipunaRanasinghe) [inline examples/Sports-headline-analyzer/Ballerina.toml]:
```suggestion
name = "sports_headline_analyzer"
```

File: examples/Sports-headline-analyzer/main.bal [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    io:print("\nTraining...");
+    while (fineTuneJob.status == "running") {
+        fineTuneJob = check openAIFinetunes->/fine_tuning/jobs/[fineTuneJobId].get();
+        io:print(".");
+        runtime:sleep(1);
+    }
+
+    if (fineTuneJob.status != "succeeded") {
```
Comment (NipunaRanasinghe) [inline examples/Sports-headline-analyzer/main.bal]:
```suggestion
    if fineTuneJob.status != "succeeded" {
```

File: examples/Sports-headline-analyzer/main.bal [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    io:println("\nFine-tuning job details: ");
+    io:println("Fine-tuned Model: ", fineTuneJob.fine_tuned_model);
+    io:println("Model: ", fineTuneJob.model);
+    io:println("Fine-tuning job completed successfully.");
+
+    finetunes:DeleteFileResponse deleteFileResponse =
+        check openAIFinetunes->/files/[trainingFileId].delete();
+    if (deleteFileResponse.deleted == true) {
```
Comment (NipunaRanasinghe) [inline examples/Sports-headline-analyzer/main.bal]:
```suggestion
    if (deleteFileResponse.deleted) {
```

File: examples/Sports-headline-analyzer/Ballerina.toml [hunk 9]
```diff
@@ -0,0 +1,14 @@
+[package]
+org = "kanishka"
```
Comment (NipunaRanasinghe) [inline examples/Sports-headline-analyzer/Ballerina.toml]:
```suggestion
org = "wso2"
```

**Review by NipunaRanasinghe — CHANGES_REQUESTED** [review]

File: examples/Sarcastic-bot/Ballerina.toml (hunk 4 above)
Comment (KATTA-00) [reply examples/Sarcastic-bot/Ballerina.toml]: noted

File: examples/Sarcastic-bot/Dependencies.toml
Comment (KATTA-00) [reply examples/Sarcastic-bot/Dependencies.toml:1]: noted

File: examples/README.md [hunk 10]
```diff
@@ -1,14 +1,20 @@
...
-1. 
-2. 
+1. [Sarcastic Bot](https://github.com/ballerina-platform/module-ballerinax-openai.finetunes/tree/main/examples/Sarcastic-bot) - Fine-tune the GPT-3.5-turbo model to generate sarcastic responses 
+
+2. [Sports Headline Analyzer](https://github.com/ballerina-platform/module-ballerinax-openai.finetunes/tree/main/examples/Sports-headline-analyzer) - Fine-tune the GPT-4o-mini model to extract structured information (player, team, sport, and gender) from sports headlines.
 
 ## Prerequisites
 
-[//]: # (TODO: Add prerequisites)
+1. Generate a API as described in the [Setup guide](https://central.ballerina.io/ballerinax/openai.finetunes/latest#setup-guide).
```
Comment (NipunaRanasinghe) [inline examples/README.md]:
```suggestion
1. Generate an API key as described in the [Setup guide](https://central.ballerina.io/ballerinax/openai.finetunes/latest#setup-guide).
```

File: examples/Sarcastic-bot/.gitignore
Comment (NipunaRanasinghe) [inline examples/Sarcastic-bot/.gitignore:1]: Do we need this .gitignore as we already have one under the repository root?

File: examples/Sports-headline-analyzer/Sports headline analyzer.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
+
+## Sports Headlines Analyzer
```
Comment (NipunaRanasinghe) [inline examples/Sports-headline-analyzer/Sports headline analyzer.md]: We should use title case in all the headers. Please fix all the similar places
```suggestion
## Sports headlines analyzer
```

File: examples/Sports-headline-analyzer/Sports headline analyzer.md [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+### 2. Configuration
+
+Create a `Config.toml` file in the example's root directory as follows:
+
+```bash
+token = "<API key>"
+```
+
+## Run the Example
```
Comment (NipunaRanasinghe) [inline examples/Sports-headline-analyzer/Sports headline analyzer.md]: ditto
```suggestion
## Run the example
```

File: examples/Sarcastic-bot/.gitignore
Comment (KATTA-00) [reply examples/Sarcastic-bot/.gitignore:1]: No, I think, it was automatically generated when creating a project. let me remove them

File: examples/Sarcastic-bot/Sarcastic bot.md [hunk 13]
```diff
@@ -0,0 +1,26 @@
+
+## Sarcastic Bot
```
Comment (NipunaRanasinghe) [inline examples/Sarcastic-bot/Sarcastic bot.md]:
```suggestion
## Sarcastic bot
```

File: README.md [hunk 14]
```diff
@@ -22,7 +22,9 @@
 
 The `OpenAI Finetunes` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/module-ballerinax-openai.finetunes/tree/main/examples/), covering the following use cases:
 
-[//]: # (TODO: Add examples)
+1. [Sarcastic Bot](https://github.com/ballerina-platform/module-ballerinax-openai.finetunes/tree/main/examples/Sarcastic-bot) - Fine-tune the GPT-3.5-turbo model to generate sarcastic responses
```
Comment (NipunaRanasinghe) [inline README.md]:
```suggestion
1. [Sarcastic bot](https://github.com/ballerina-platform/module-ballerinax-openai.finetunes/tree/main/examples/Sarcastic-bot) - Fine-tune the GPT-3.5-turbo model to generate sarcastic responses 
```

File: README.md [hunk 15]
```diff
@@ -22,7 +22,9 @@
 
 The `OpenAI Finetunes` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/module-ballerinax-openai.finetunes/tree/main/examples/), covering the following use cases:
 
-[//]: # (TODO: Add examples)
+1. [Sarcastic Bot](https://github.com/ballerina-platform/module-ballerinax-openai.finetunes/tree/main/examples/Sarcastic-bot) - Fine-tune the GPT-3.5-turbo model to generate sarcastic responses 
+
+2. [Sports Headline Analyzer](https://github.com/ballerina-platform/module-ballerinax-openai.finetunes/tree/main/examples/Sports-headline-analyzer) - Fine-tune the GPT-4o-mini model to extract structured information (player, team, sport, and gender) from sports headlines.
```
Comment (NipunaRanasinghe) [inline README.md]:
```suggestion
2. [Sports headline analyzer](https://github.com/ballerina-platform/module-ballerinax-openai.finetunes/tree/main/examples/Sports-headline-analyzer) - Fine-tune the GPT-4o-mini model to extract structured information (player, team, sport, and gender) from sports headlines.
```

File: README.md [hunk 16]
```diff
@@ -22,7 +22,9 @@
 
 The `OpenAI Finetunes` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/module-ballerinax-openai.finetunes/tree/main/examples/), covering the following use cases:
 
-[//]: # (TODO: Add examples)
```
Comment (NipunaRanasinghe) [inline README.md:102]: Need to update the same in the `Module.md` and `Package.md` as well

**Review by NipunaRanasinghe — CHANGES_REQUESTED** [review]

**Review by NipunaRanasinghe — DISMISSED** [review]

File: ballerina/Dependencies.toml [hunk 17]
```diff
@@ -318,7 +318,7 @@ modules = [
 [[package]]
 org = "ballerinax"
 name = "openai.finetunes"
-version = "1.0.5"
+version = "1.0.7"
```
Comment (NipunaRanasinghe) [inline ballerina/Dependencies.toml]:
```suggestion
version = "1.0.5"
```

File: ballerina/Dependencies.toml [hunk 18]
```diff
@@ -107,7 +107,7 @@ version = "0.0.0"
 [[package]]
 org = "ballerina"
 name = "jwt"
-version = "2.12.1"
+version = "2.11.0"
```
Comment (NipunaRanasinghe) [inline ballerina/Dependencies.toml]:
```suggestion
version = "2.12.1"
```

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**